operators/src/processing/temporal_raster_aggregation.rs

```diff
@@ -0,0 +1,1074 @@
+            source,
+            tiling_specification: context.tiling_specification(),
+        };
+
+        Ok(initialized_operator.boxed())
+    }
+}
+
+pub struct InitializedTemporalRasterAggregation {
+    aggregation_type: AggregationType,
+    window: TimeStep,
+    ignore_no_data: bool,
+    source: Box<InitializedRasterOperator>,
+    result_descriptor: RasterResultDescriptor,
+    tiling_specification: TilingSpecification,
+}
+
+impl InitializedOperator<RasterResultDescriptor, TypedRasterQueryProcessor>
+    for InitializedTemporalRasterAggregation
+{
+    fn result_descriptor(&self) -> &RasterResultDescriptor {
+        &self.result_descriptor
+    }
+
+    fn query_processor(&self) -> Result<TypedRasterQueryProcessor> {
+        let source_processor = self.source.query_processor()?;
+
+        let res = call_on_generic_raster_processor!(
+            source_processor, p =>
+            TemporalRasterAggregationProcessor::new(
+                self.aggregation_type,
+                self.window,
+                self.ignore_no_data,
+                p,
+                self.tiling_specification,
+                self.source.result_descriptor().no_data_value
+            )
+            .boxed().into()
+        );
+
+        Ok(res)
+    }
+}
+
+pub struct TemporalRasterAggregationProcessor<Q, P>
+where
+    Q: RasterQueryProcessor<RasterType = P>,
+    P: Pixel,
+{
+    aggregation_type: AggregationType,
+    window: TimeStep,
+    ignore_no_data: bool,
+    source: Q,
+    tiling_specification: TilingSpecification,
+    no_data_value: Option<P>,
+}
+
+impl<Q, P> TemporalRasterAggregationProcessor<Q, P>
+where
+    Q: RasterQueryProcessor<RasterType = P>,
+    P: Pixel,
+{
+    fn new(
+        aggregation_type: AggregationType,
+        window: TimeStep,
+        ignore_no_data: bool,
+        source: Q,
+        tiling_specification: TilingSpecification,
+        no_data_value: Option<f64>,
+    ) -> Self {
+        Self {
+            aggregation_type,
+            window,
+            ignore_no_data,
+            source,
+            tiling_specification,
+            no_data_value: no_data_value.map(P::from_),
+        }
+    }
+}
+
+impl<Q, P> RasterQueryProcessor for TemporalRasterAggregationProcessor<Q, P>
+where
+    P: Pixel,
+    Q: RasterQueryProcessor<RasterType = P>,
+{
+    type RasterType = P;
+
+    fn raster_query<'a>(
+        &'a self,
+        query: crate::engine::QueryRectangle,
+        ctx: &'a dyn crate::engine::QueryContext,
+    ) -> Result<futures::stream::BoxStream<'a, Result<RasterTile2D<Self::RasterType>>>> {
+        match self.aggregation_type {
+            AggregationType::Min => {
+                if self.ignore_no_data {
+                    let spec = TemporalRasterAggregationSubQuery {
+                        fold_fn: no_data_ignoring_fold_future::<P, MinIgnoreNoDataAccFunction>,
+                        no_data_value: self.no_data_value,
+                        initial_value: P::max_value(),
```

Comment:
I guess you should add tests with empty raster streams. Do we get a very large minimum there?

Reply:
there are no empty raster stream, rather a stream with no data in a query rectangle would get a no data tile which would lead to no data output or, in the case of ignore no data, max or min values. but then it's the users fault for ignoring the no data.

Reply:
Okay, and does it work with just one empty tile?

Reply:
e1f6564563051a9654ea86b41c8379f18c66e178